tasks/main.yml

```diff
@@ -1,20 +1,57 @@
 ---
-- name: Install targetcli packages
+- name: Install targetcli packages on RedHat/CentOS/Fedora
   yum:
     name: 'targetcli'
     state: 'installed'
+  when: 'ansible_os_family == "RedHat"'
 
-- name: Start and enable targetcli service
+- name: Install targetcli packages on SUSE 15.x / openSUSE 15.x or higher
+  package:
+    name: 'python3-targetcli-fb'
+    state: 'installed'
+    disable_recommends: true
+  when: '(ansible_os_family == "Suse" or ansible_os_family == "openSUSE Leap") and ansible_distribution_major_version|int >= 15'
+
+- name: Start and enable targetcli service on RedHat/CentOS/Fedora
   service:
     name: 'target'
     enabled: true
     state: 'started'
+  when: 'ansible_os_family == "RedHat"'
+
+- name: Start and enable targetcli service on SUSE 15.x / openSUSE 15.x or higher
+  service:
+    name: 'targetcli'
+    enabled: true
+    state: 'started'
+  when: '(ansible_os_family == "Suse" or ansible_os_family == "openSUSE Leap") and ansible_distribution_major_version|int >= 15'
+
+- name: Check if firewalld is installed
+  command:
+  args:
+    cmd: 'rpm -q firewalld'
+    warn: false
+  register: check_for_firewalld
+  failed_when: false
+  changed_when: false
+  check_mode: false
+
+- name: Check if firewalld is running
+  command:
+  args:
+    cmd: 'systemctl is-active firewalld'
+    warn: false
+  register: check_for_running_firewalld
+  failed_when: false
+  changed_when: false
+  check_mode: false
 
 - name: Enable port 3260/tcp on firewalld
   firewalld:
     port: '3260/tcp'
     permanent: true
     state: 'enabled'
+  when: '"firewalld is not installed" not in check_for_firewalld.stdout and "inactive" not in check_for_running_firewalld.stdout'
```

Comment:
same feedback as in PR #3 applies here - and this can be adjusted to just check if the `firewalld.service` is running

Reply:
Sorry, this PR is based on #3 and thus contains the same set of changes. Once #3 is merged, I can rebased this PR.